src/ota_flash.c

```diff
@@ -187,11 +187,11 @@ MG_IRAM void mg_ota_boot(void) {
     (void) tmpsector;
     for (ofs = 0; ofs < max; ofs += ss) {
       // mg_flash_erase(tmpsector);
-      mg_flash_write(tmpsector, partition1 + ofs, ss);
+      mg_flash_write(tmpsector, partition1 + ofs + MG_FLASH_OFFSET, ss);
       // mg_flash_erase(partition1 + ofs);
-      mg_flash_write(partition1 + ofs, partition2 + ofs, ss);
+      mg_flash_write(partition1 + ofs, partition2 + ofs + MG_FLASH_OFFSET, ss);
       // mg_flash_erase(partition2 + ofs);
-      mg_flash_write(partition2 + ofs, tmpsector, ss);
+      mg_flash_write(partition2 + ofs, tmpsector + MG_FLASH_OFFSET, ss);
     }
     mg_device_reset();
   }
```

Comment:
Did you actually try to swap partitions and succeed ?
Log calls in mg_flash_erase() will eventually mean you are calling flash that has been erased...

Reply:
Before we start swapping the partitions, we disable the logs, so the macro has no effect.

Reply:
The macro always has an effect, code has been compiled in with log enabled so that macro generated some code.
Does that RAM code actually check for the log level and then skips jumping to the log function, whatever it is ? (The user can change it)
Then that is cool and quite clever.
Bear in mind that all those strings are quite likely using valuable RAM space, nevetheless. (for the same reason)

Reply:
Yes, the macro will first check the log level value and then decides if it the logging function will be executed. In the case of MG_LL_NONE (we set logging to this value in mg_ota_boot, right before swapping), no logging function will be executed.